Find-Package.NuGet.ArgumentCompleters.ps1

```diff
@@ -0,0 +1,84 @@
+<#PSScriptInfo
+.VERSION 0.0.1
+.AUTHOR brywang-msft
+.COPYRIGHT (c) Microsoft Corporation
+.GUID b4267027-4764-47fa-928f-61e7f6db43c3
+.TAGS Find-Package, PackageManagement, TabExpansion2, Register-ArgumentCompleter
+.LICENSEURI http://www.apache.org/licenses/LICENSE-2.0
+.PROJECTURI https://github.com/oneget/NuGetProvider
+#>
+
+<#
+.Synopsis
+	Argument completers for Find-Package -Name.
+.Description
+	The script registers argument completers for Find-Package -Name when a single source is registered OR -Source is specified. All credential parameters should be specified before pressing <tab>.
+	Installation:
+	* Using PowerShell v5+:
+		Invoke Find-Package.NuGet.ArgumentCompleters.ps1 (or add it to your profile)
+	* TabExpansionPlusPlus:
+		Find-Package.NuGet.ArgumentCompleters.ps1 should be placed in the same directory as TabExpansionPlusPlus. Optionally, invoke Find-Package.NuGet.ArgumentCompleters.ps1 (or add it to your profile)
+	* TabExpansion2.ps1
+		Find-Package.NuGet.ArgumentCompleters.ps1 should be added to the path. Optionally, AFTER invoking TabExpansion2.ps1, invoke Find-Package.NuGet.ArgumentCompleters.ps1 (or add it to your profile)
+#>
+Register-ArgumentCompleter -CommandName Find-Package -ParameterName Name -ScriptBlock {
+	param($commandName, $parameterName, $wordToComplete, $commandAst, $boundParameters)
+
+	# Check that the loaded PackageManagement module is new enough
+	if (-not ("Microsoft.PackageManagement.NuGetProvider.RequestWrapper" -as [Type])) {
+		return
+	}
+
+	# This is a workaround until we can figure out how to properly page results. Currently argument completers seem to fully enumerate the results, so we have to limit the number of results ourselves.
+	# Change this if you want more results, but be aware that the script's signing becomes invalid.
+	$MAX_COMPLETION_RESULTS = 20
+
+
+    $location = $null # String
+    $credentialUserName = $null # String
+    $credentialUserPassword = $null # SecureString
+    $proxy = $null # IWebProxy
+    $headers = $null # string[]
```

Comment:
Consider adding type qualifier to the variables if these expected types.

Reply:
Adding

---
In reply to: [154492184](https://github.com/OneGet/NuGetProvider/pull/25#discussion_r154492184) [](ancestors = 154492184)